dm/dm/master/server.go

```diff
@@ -1436,8 +1465,20 @@ func (s *Server) OperateLeader(ctx context.Context, req *pb.OperateLeaderRequest
 	}, nil
 }
 
-func (s *Server) generateSubTask(ctx context.Context, task string, errCnt, warnCnt int64) (*config.TaskConfig, []*config.SubTaskConfig, error) {
+func (s *Server) generateSubTask(
+	ctx context.Context,
+	task string,
+	errCnt,
+	warnCnt int64,
+	cliArgs *config.TaskCliArgs,
+) (*config.TaskConfig, []*config.SubTaskConfig, error) {
 	cfg := config.NewTaskConfig()
+	// bypass the meta check by set any value. If start-time is specified, DM-worker will not use meta field.
+	if cliArgs != nil && cliArgs.StartTime != "" {
+		for _, inst := range cfg.MySQLInstances {
+			inst.Meta = &config.Meta{BinLogName: cliArgs.StartTime}
```

Comment:
why not add a new field in meta? this is too trick

Reply:
this structure is saved in etcd to pass to DM-worker, its changing will cause downgrade problems.

Reply:
yes i konw, but we already have a lot of new configs in stubTaskConfig such as `CollationCompatible` 😂
so it's now difficult for user to downgrade version automatically.

Reply:
the difference with `CollationCompatible` is `CollationCompatible` can be simply removed when TaskConfigForDowngrade, but removing "a new field in meta" will left meta empty and can't pass valid check, so I still need to write a dummy message back.

Reply:
i am curious that why we need put start-time in  meta? seems worker can get stat-time from etcd `cliArgs`?

Reply:
> i am curious that why we need put start-time in meta? seems worker can get stat-time from etcd `cliArgs`?

line 1476 "bypass the meta check by set any value"